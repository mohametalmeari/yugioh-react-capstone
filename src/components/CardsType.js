import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { ArrowIcon } from '../icons';
import monsterTypeImg from '../images/1.png';
import otherTypeImg from '../images/2.png';

const CardsType = ({ typeId, typeName, typeCount }) => (
  <NavLink
    to={`type-${typeId}`}
  >
    <div className="cards-type">
      <img data-testid="img-2" className="type-img" src={typeName.includes('Monster') ? monsterTypeImg : otherTypeImg} alt="Type" />
      <ArrowIcon />
      <div className="streched-gap" />
      <h3 className="type-title">{typeName}</h3>
      <p className="type-count">
        {typeCount}
        {' '}
        cards
      </p>
    </div>
  </NavLink>
);
CardsType.propTypes = {
  typeId: PropTypes.string.isRequired,
  typeName: PropTypes.string.isRequired,
  typeCount: PropTypes.number.isRequired,
};

export default CardsType;
