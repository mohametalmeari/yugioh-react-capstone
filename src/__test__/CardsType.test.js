import { render, screen, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import CardsType from '../components/CardsType';

describe('Cards Type Component', () => {
  afterEach(() => cleanup());
  test('renders type', () => {
    render(
      <MemoryRouter>
        <CardsType
          key="777"
          typeId="777"
          typeName="Spell Card"
          typeCount={50}
        />
      </MemoryRouter>,
    );
    expect(screen.getByText(/Spell Card/i)).toBeInTheDocument();
    expect(screen.getByText(/50 cards/i)).toBeInTheDocument();
  });
  test('renders type image', () => {
    render(
      <MemoryRouter>
        <CardsType
          key="999"
          typeId="999"
          typeName="Spell Card"
          typeCount={50}
        />
      </MemoryRouter>,
    );
    expect(screen.getByTestId('img-2')).toBeInTheDocument();
  });
  test('renders Type Snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <CardsType
          key="888"
          typeId="888"
          typeName="Spell Card"
          typeCount={50}
        />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
