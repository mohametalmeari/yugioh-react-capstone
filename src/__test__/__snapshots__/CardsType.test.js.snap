// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Type Component renders Type Snapshot 1`] = `
<div>
  <a
    class=""
    href="/type-888"
  >
    <div
      class="cards-type"
    >
      <img
        alt="Type"
        class="type-img"
        data-testid="img-2"
        src="2.png"
      />
      <svg
        class="w-6 h-6"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.75 15l3-3m0 0l-3-3m3 3h-7.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <div
        class="streched-gap"
      />
      <h3
        class="type-title"
      >
        Spell Card
      </h3>
      <p
        class="type-count"
      >
        50
         
        cards
      </p>
    </div>
  </a>
</div>
`;
